### PR TITLE
Replace cpliakas/git-wrapper with gitonomy/gitlib

### PR DESCRIPTION
**Pull request by pheeque1**
This PR fixes issue #151

### CODE REVIEW

**Comment by ostark** [comment]
@pheeque1 is this finished and tested?

File: src/Services/Git/GitonomyClient.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		}, $localBranches);
+	}
+
+	public function getRemotes(?string $for = 'push'): array
+	{
+		if (!in_array($for, ['push', 'pull'], true)) {
+			throw new LogicException(
+				sprintf(
+					'Argument 1 passed to %s must be "pull" or "push", %s given.',
+					'fortrabbit\Copy\Services\Git\Client::getRemotes()',
```
Comment (ostark) [inline src/Services/Git/GitonomyClient.php:66]: please update

**Review by ostark — APPROVED** [review]
LGTM, just requesting a tiny change regarding a leftover

File: src/Services/Git/GitonomyClient.php (hunk 1 above)
Comment (pheeque1) [reply src/Services/Git/GitonomyClient.php]: I have pushed an update that references the correct class.

File: src/Services/Git/GitonomyClient.php [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+		// Split upstream/branch and return upstream only
+		return explode('/', $result)[0];
+	}
+
+	public function checkout(string $branch)
+	{
+		$this->repository->run(GitCommand::CHECKOUT, [$branch]);
+	}
+
+	public function addRemote(string $name, ?string $url)
```
Comment (ostark) [inline src/Services/Git/GitonomyClient.php:102]: @pheeque1 
I tested it and notice the remote was not added.
That's the reason why :-)

File: src/Services/Git/GitonomyClient.php [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+			$argsOrOptions = [
+				'-a',
+				'-m ' . $argsOrOptions[0],
+			];
+		}
+
+		return $this->run(GitCommand::COMMIT, $argsOrOptions);
+	}
+
+	public function streamOutput(bool $streamOutput = true): void
```
Comment (ostark) [inline src/Services/Git/GitonomyClient.php:154]: No need to implement it, if it does nothing

Comment (pheeque1) [reply src/Services/Git/GitonomyClient.php]: Removed.

File: src/Services/Git/GitonomyClient.php (hunk 2 above)
Comment (pheeque1) [reply src/Services/Git/GitonomyClient.php:102]: 🤦🏾‍♂️ Totally missed this. Implemented now.
Thanks.

File: src/Services/Git.php [hunk 4]
```diff
@@ -93,54 +80,15 @@ public function getLocalBranches(): array
...
-        } catch (GitException) {
-            return [];
-        }
-
-        foreach ($remotes as $name => $upstreams) {
-            $remotes[$name] = $upstreams[$for];
-        }
-
-        return $remotes;
+        return $this->gitClient->getRemotes($for);
```
[the remark this answers is not in the document]
Comment (pheeque1) [reply src/Services/Git.php:83]: Fix added. Thanks 😅

**Pull request merged**